solutions/silver/usaco-645.mdx

```diff
@@ -2,159 +2,178 @@
 id: usaco-645
 source: USACO Gold 2016 Open
 title: Splitting the Field
-author: Óscar Garries, Benjamin Qi
+author: Óscar Garries, Benjamin Qi, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_split_gold_open16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_split_gold_open16.html)
 
-<LanguageSection>
+<Warning title="Weak Test Data">
 
-<CPPSection>
+Some accepted submissions may give different outputs on the following test case:
+
+The right answer should be $6$.
+
+```cpp
+// BeginCodeSnip{Test Case}
+17 1 1 1 2 1 3 2 1 2 2 2 3 3 1 3 2 3 3 3 4 3 5 4 3 4 4 4 5 5 3 5 4 5 5
+    // EndCodeSnip
+```
+</Warning>
+
+## Explanation
+
+Since the enclosures must be rectangular, there *must* be a vertical or horizontal line that separates the two. Otherwise,
+we'd have a case where one enclosure would have to overlap the other in order to enclose all the cows.
+
+![](<usaco-645/enclosures.png>)
+
+Thus, we'll perform two sweeps, one checking vertical splits, and the other checking horizontal splits. To keep track of the best answer, we'll keep
+prefix and suffix minimums and maximums to calculate each possible solution in $\mathcal{O}(1)$ time.
+
+Using binary search instead of prefix/suffix minimums and maximums is also possible.
```

Comment:
Wait can you explain the binary search part further

Reply:
sure!